### ISSUE TITLE
Hangs on Starting Continue Server

### ISSUE THREAD

**Issue by ferrantejake** [continuedev/continue#311]
**Describe the bug**

After installing the extension, I am getting a notification stating that the Continue server is starting. This message doesn't go away and I'm unable to use the extension.

**To Reproduce**

1. Install Continue extension (happens right away upon install)

**Environment**
- Operating System: Win11
- Python Version: 3.11.4
- Continue Version: ~v0.0.208~ v0.0.209

More details:
VSCode/Box details:
Version: 1.80.1 (user setup)
Commit: 74f6148eb9ea00507ec113ec51c489d6ffb4b771
Date: 2023-07-12T17:22:07.651Z
Electron: 22.3.14
ElectronBuildId: 21893604
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621

**Console logs (DevTools)**
```
[Extension Host] Websocket import
console.ts:137 [Extension Host] Using Continue version:  0.0.208
console.ts:137 [Extension Host] Killing old server...
console.ts:137 [Extension Host] Failed to kill old server: Error: spawn c:\Users\jakef\.vscode\extensions\continue.continue-0.0.208\out\fastlist.exe ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:285:19)
    at onErrorNT (node:internal/child_process:506:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
console.ts:137 [Extension Host] Setting up python env for Continue extension...
console.ts:137 [Extension Host] Running command:  python3 --version
console.ts:137 [Extension Host] Running command:  python --version
console.ts:137 [Extension Host] Python env already exists, skipping...
console.ts:137 [Extension Host] Running command:  python3 --version
console.ts:137 [Extension Host] Running command:  python --version
console.ts:137 [Extension Host] Running command:  cd "C:\Users\jakef\.continue\server" ; .\env\Scripts\activate ; pip uninstall -y continuedev
console.ts:137 [Extension Host] Running command:  cd "C:\Users\jakef\.continue\server" ; .\env\Scripts\activate ; python -m pip install --upgrade pip ; pip install -r requirements.txt
console.ts:137 [Extension Host] Running command:  python3 --version
console.ts:137 [Extension Host] Running command:  python --version
console.ts:137 [Extension Host] Starting Continue python server...
console.ts:137 [Extension Host] stdout: INFO:     Started server process [27288]
INFO:     Waiting for application startup.
console.ts:137 [Extension Host] stdout: INFO:     Application startup complete.
ERROR:    [Errno 10048] error while attempting to bind on address ('0.0.0.0', 65432): only one usage of each socket address (protocol/network address/port) is normally permitted
INFO:     Waiting for application shutdown.
console.ts:137 [Extension Host] Successfully started Continue python server
console.ts:137 [Extension Host] Continue server started
console.ts:137 [Extension Host] Creating websocket at:  ws://localhost:65432/ide/ws
console.ts:137 [Extension Host] stdout: INFO:     Application shutdown complete.
console.ts:137 [Extension Host] stdout: Task exception was never retrieved
future: <Task finished name='Task-1' coro=<Server.serve() done, defined at C:\Users\jakef\.continue\server\env\Lib\site-packages\uvicorn\server.py:61> exception=SystemExit(1)>
Traceback (most recent call last):
  File "C:\Users\jakef\.continue\server\env\Lib\site-packages\uvicorn\server.py", line 159, in startup
    server = await loop.create_server(
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\asyncio\base_events.py", line 1525, in create_server
    raise OSError(err.errno, 'error while attempting '
OSError: [Errno 10048] error while attempting to bind on address ('0.0.0.0', 65432): only one usage of each socket address (protocol/network address/port) is normally permitted

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\jakef\.continue\server\env\Lib\site-packages\nest_asyncio.py", line 35, in run
    return loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jakef\.continue\server\env\Lib\site-packages\nest_asyncio.py", line 84, in run_until_complete
    self._run_once()
  File "C:\Users\jakef\.continue\server\env\Lib\site-packages\nest_asyncio.py", line 120, in _run_once
    handle._run()
  File "C:\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Python311\Lib\asyncio\tasks.py", line 350, in __wakeup
    self.__step()
  File "C:\Users\jakef\.continue\server\env\Lib\site-packages\nest_asyncio.py", line 196, in step
    step_orig(task, exc)
  File "C:\Python311\Lib\asyncio\tasks.py", line 267, in __step
    result = coro.send(None)
             ^^^^^^^^^^^^^^^
  File "C:\Users\jakef\.continue\server\env\Lib\site-packages\uvicorn\server.py", line 76, in serve
    await self.startup(sockets=sockets)
  File "C:\Users\jakef\.continue\server\env\Lib\site-packages\uvicorn\server.py", line 169, in startup
    sys.exit(1)
SystemExit: 1
console.ts:137 [Extension Host] Successfully started Continue python server
8
console.ts:137 [Extension Host] Websocket not yet open, trying again...
index.js:560 Starting React
2
console.ts:137 [Extension Host] Running onLoad
console.ts:137 [Extension Host] Websocket not yet open, trying again...
undefined.gif:1 
 Failed to load resource: the server responded with a status of 404 ()
4
console.ts:137 [Extension Host] Websocket not yet open, trying again...
semanticTokensProviderStyling.ts:108 Overlapping semantic tokens detected at lineNumber 17, column 33
150
console.ts:137 [Extension Host] Websocket not yet open, trying again...
DevTools failed to load source map: Could not load content for https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jakef/.vscode/extensions/continue.continue-0.0.208/react-app/dist/assets/index.js.map: Connection error: net::ERR_NAME_NOT_RESOLVED
171
console.ts:137 [Extension Host] Websocket not yet open, trying again...
```

**Comment by vankeer** [comment]
I have the same issue when installing the extension on Ubuntu 22.04

It's fixed when I disable it, restart Code, then enable again.

However, I cannot get the server to load the `~/.continue/config.py` file it seems. With a custom `default_model` set to `ggml`, it still uses the default GPT API. Also, there is only the `/test` command available. When I change the command's description in config.py, it's not updated. So the config file clearly isn't picked up. I notice the server keeps running after closing Code (that in itself is a separate bug probably). When I kill the server process and restart Code, often times the "Starting Continue Server" message hangs, but not always. The updated config.py isn't picked up.

I feel like all these issues are related. I can only get this extension to work when building the it in development mode & running the dev server manually. This is a really nice project, but probably there should be a disclaimer in the readme file that this plugin is still very much in active development.

**Comment by TKlerx** [comment]
Same here. I also changed to ggml, but the extension still uses OpenAI. (tried versions 0.0.211 - 0.0.215)

**Comment by sestinj** [comment]
@TKlerx , @vankeer I've since found the solution to the config not being acknowledged (errors were being swallowed, resulting in falling back to a default config), and the latest version on the VS Code Extension Marketplace will solve this.

It is true that the server runs after closing VS Code—that issue can be tracked here: #322 

And a couple of other notes, @vankeer: Since you are running the server manually, this will break the default config imports (they need an additional continuedev.src. appended in front), but you'll now see a traceback in the UI if anything is wrong with config.py. And it's a fair point to add a disclaimer. We're chipping away at these loading issues, but it might help set expectations.

**Comment by TKlerx** [comment]
It is working now with my local model. Thank you!

**Comment by sestinj** [comment]
@ferrantejake @vankeer An update for those watching this issue:

1. I'm working [here](https://github.com/continuedev/continue/issues/320) on a solution to bundle the python code as an executable, which I expect to solve this issue
2. For the time being, I've added documentation [here](https://github.com/continuedev/continue/tree/main/extension#troubleshooting) on how to troubleshoot this problem

**Comment by karthik5699** [comment]
I’m having trouble loading Continue Server and when I checked the logs, the problem seems to be with starting MeiliSearch.

**Comment by sestinj** [comment]
@karthik5699 I believe I have solved this error as of yesterday. If not, let me know and I'll create a new issue for it.

**Comment by sestinj** [comment]
@ferrantejake @vankeer I've now merged the update that packages the server as an executable, so I would expect these issues to be resolved. Please let me know whether the problem still exists. Excited to get this working for you!

**Comment by ferrantejake** [comment]
@sestinj Thanks for your attention and commitment to making this work! I just tried the latest (apparently released only 30 mins ago!). It looks like the issue above is resolved! 

In testing the latest, I was still not able to get past starting the extension. I have made a ticket for it here: https://github.com/continuedev/continue/issues/345

I'll close this ticket since the root issue appears resolved.
